pkg/network/ebpf/c/protocols/redis/decoding.h

```diff
@@ -209,13 +235,47 @@ static __always_inline void redis_batch_enqueue_wrapper(conn_tuple_t *tuple, red
 
 // Processes Redis response messages and validates their format.
 // Handles error responses and command-specific response types.
+static __always_inline redis_error_t map_redis_error_prefix(const char *prefix) {
+    if (bpf_memcmp(prefix, "ERR", 3) == 0) return REDIS_ERR_ERR;
+     if (bpf_memcmp(prefix, "WRONGTYPE", 9) == 0) return REDIS_ERR_WRONGTYPE;
+     if (bpf_memcmp(prefix, "NOAUTH", 6) == 0) return REDIS_ERR_NOAUTH;
+     if (bpf_memcmp(prefix, "NOPERM", 6) == 0) return REDIS_ERR_NOPERM;
+     if (bpf_memcmp(prefix, "BUSY", 4) == 0) return REDIS_ERR_BUSY;
+     if (bpf_memcmp(prefix, "NOSCRIPT", 8) == 0) return REDIS_ERR_NOSCRIPT;
+     if (bpf_memcmp(prefix, "LOADING", 7) == 0) return REDIS_ERR_LOADING;
+     if (bpf_memcmp(prefix, "READONLY", 8) == 0) return REDIS_ERR_READONLY;
+     if (bpf_memcmp(prefix, "EXECABORT", 9) == 0) return REDIS_ERR_EXECABORT;
+     if (bpf_memcmp(prefix, "MASTERDOWN", 10) == 0) return REDIS_ERR_MASTERDOWN;
+     if (bpf_memcmp(prefix, "MISCONF", 7) == 0) return REDIS_ERR_MISCONF;
+     if (bpf_memcmp(prefix, "CROSSSLOT", 9) == 0) return REDIS_ERR_CROSSSLOT;
+     if (bpf_memcmp(prefix, "TRYAGAIN", 8) == 0) return REDIS_ERR_TRYAGAIN;
+     if (bpf_memcmp(prefix, "ASK", 3) == 0) return REDIS_ERR_ASK;
+     if (bpf_memcmp(prefix, "MOVED", 5) == 0) return REDIS_ERR_MOVED;
+     if (bpf_memcmp(prefix, "CLUSTERDOWN", 11) == 0) return REDIS_ERR_CLUSTERDOWN;
+     if (bpf_memcmp(prefix, "NOREPLICAS", 10) == 0) return REDIS_ERR_NOREPLICAS;
+     if (bpf_memcmp(prefix, "OOM", 3) == 0) return REDIS_ERR_OOM;
+     if (bpf_memcmp(prefix, "NOQUORUM", 8) == 0) return REDIS_ERR_NOQUORUM;
+     if (bpf_memcmp(prefix, "BUSYKEY", 7) == 0) return REDIS_ERR_BUSYKEY;
+     if (bpf_memcmp(prefix, "UNBLOCKED", 9) == 0) return REDIS_ERR_UNBLOCKED;
+     if (bpf_memcmp(prefix, "WRONGPASS", 9) == 0) return REDIS_ERR_WRONGPASS;
+     if (bpf_memcmp(prefix, "INVALIDOBJ", 7) == 0) return REDIS_ERR_INVALIDOBJ;
+    return REDIS_ERR_UNKNOWN;
+}
+
 static void __always_inline process_redis_response(pktbuf_t pkt, conn_tuple_t *tup, redis_transaction_t *transaction) {
     char first_byte;
     if (pktbuf_load_bytes_from_current_offset(pkt, &first_byte, sizeof(first_byte)) < 0) {
         return;
     }
     if (first_byte == RESP_ERROR_PREFIX) {
         transaction->is_error = true;
+        char error_prefix[MAX_ERROR_SIZE] = {};
+        bool got_error = extract_redis_error_prefix(pkt, error_prefix, sizeof(error_prefix));
+        if (got_error) {
+            transaction->error = map_redis_error_prefix(error_prefix);
+        } else {
+            transaction->error = REDIS_ERR_UNKNOWN;
+        }
```

Comment:
With my previous comment, I think the condition is redundant, as eventually `transaction->error = map_redis_error_prefix` should suffice, as the default behavior is `REDIS_ERR_UNKNOWN`